src/test/java/org/cactoos/text/SplitTest.java

```diff
@@ -92,4 +103,99 @@ public void splitTextWithTextRegex() throws Exception {
             new TextHasString("Split")
         ).affirm();
     }
+
+    @Test
+    public void splitTextPositiveLimit() throws Exception {
+        new Assertion<>(
+            "Must split a string with limited length",
+            new LengthOf(
+                new Split("boo:and:foo", ":", 2)
+            ).intValue(),
+            new IsEqual<>(2)
+        ).affirm();
+    }
+
+    @Test
+    public void splitTextNegativeLimit() throws Exception {
+        new Assertion<>(
+            "Must split a string with limited length",
+            new LengthOf(
+                new Split("boo:and:foo", ":", -1)
+            ).intValue(),
+            // @checkstyle MagicNumberCheck (1 line)
+            new IsEqual<>(3)
+        ).affirm();
+    }
+
+    @Test
+    public void splitTextZeroLimit() throws Exception {
+        new Assertion<>(
+            "Must split a string with limited length",
+            new LengthOf(
+                new Split("boo:and:foo", ":", 0)
+            ).intValue(),
+            // @checkstyle MagicNumberCheck (1 line)
+            new IsEqual<>(3)
+        ).affirm();
+    }
+
+    @Test
+    public void splitTextLengthLimitWithTextItem() throws Exception {
+        new Assertion<>(
+            "Must split a text with limited length",
+            new LengthOf(
+                new Split(new TextOf("foo:and:boo"), ":", 2)
+            ).intValue(),
+            new IsEqual<>(2)
+        ).affirm();
+    }
+
+    @Test
+    public void splitTextLengthLimitWithTextRegex() throws Exception {
+        new Assertion<>(
+            "Must split a string with limited length and text regex",
+            new LengthOf(
+                new Split("bo:and:fo", new TextOf(":"), 2)
+            ).intValue(),
+            new IsEqual<>(2)
+        ).affirm();
+    }
+
+    @Test
+    public void splitTextLengthLimitWithTextItemAndRegex() throws Exception {
+        new Assertion<>(
+            "Must split a text with limited length and text regex",
+            new LengthOf(
+                new Split(new TextOf("fo:and:bo"), new TextOf(":"), 2)
```

Comment:
@victornoel It is better to check for the elements, not the size of the array

Reply:
@paulodamaso I agree, I duplicated existing tests and didn't think to much about it. I will rewrite all the tests